src/wmtk/quadrature/ClippedQuadrature.cpp

```diff
@@ -0,0 +1,77 @@
+#include "ClippedQuadrature.h"
+
+#include <wmtk/utils/PolygonClipping.h>
+#include "TriangleQuadrature.h"
+
+namespace wmtk {
+
+void ClippedQuadrature::clipped_triangle_box_quadrature(
+    const int order,
+    const TriangleVertices& triangle,
+    const Eigen::AlignedBox2d& box,
+    Quadrature& quadr,
+    Quadrature* tmp)
```

Comment:
The problem with the opaque object is that it's easy to forget and create one object per call, or be tempted to call it from multiple threads, etc. But I agree this would be a nicer interface.